### PR TITLE
Add readme and improve a little the sample

### LINKED ISSUE #9: Documentation in README.md
I know this project is still in development and not usable yet, but it looks very attractive, especially for people coming from **Scala** and **Circe**, so why don't put some effort into creating some sort of documentation, at least in the `README.md`. I believe it would help attract more developers to use and contribute to it.

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #9 
Related to https://github.com/47deg/helios/issues/28

### CODE REVIEW

File: helios-sample/src/main/kotlin/helios/sample/sample.kt [hunk 1]
```diff
@@ -27,57 +28,67 @@ const val companyJsonString = """
   ]
 }"""
 
-fun main(args: Array<String>) {
+  @JvmStatic
+  fun main(args: Array<String>) {
 
-  val companyJson: Json = Json.parseUnsafe(companyJsonString)
+    val companyJson: Json = Json.parseUnsafe(companyJsonString)
```
Comment (nomisRev) [inline helios-sample/src/main/kotlin/helios/sample/sample.kt]: This should all stay 2 spaced

File: README.md [hunk 2]
```diff
@@ -1 +1,60 @@
-EXPERIMENTAL WIP Json library based on a port of the [Jawn Parser](https://github.com/non/jawn) built on Arrow.
+# Helios
+
+Json library based on a port of the [Jawn Parser](https://github.com/non/jawn) built on Arrow.
+
+## How to use it
```
Comment (nomisRev) [inline README.md]: Mention here that `jitpack` needs to be added

Comment (AdrianRaFo) [reply README.md]: this one https://github.com/jitpack/jitpack.io?

File: helios-sample/src/main/kotlin/helios/sample/sample.kt (hunk 1 above)
Comment (AdrianRaFo) [reply helios-sample/src/main/kotlin/helios/sample/sample.kt]: It's 2 spaced but it's in a object now to be able to run the sample

**Comment by AdrianRaFo** [comment]
@nomisRev I added your comments let me know if works for you

File: README.md [hunk 3]
```diff
@@ -1 +1,64 @@
...
+  companion object
+}
+```
+
+The `@json` annotation will provide the decoder and encoder for that data class, so we are able to
+
+### Decode
+
+```kotlin
+val jsonStr =
```
Comment (raulraja) [inline README.md]: We should use ank in these snippets

File: README.md [hunk 4]
```diff
@@ -1 +1,64 @@
...
+### Decode
+
+```kotlin
+val jsonStr = 
+"""{
+     "name": "Simon",
+     "age": 30
+   }"""
+    
+val jsonFromString : Json = Json.parseUnsafe(companyJsonString)
```
Comment (raulraja) [inline README.md]: Not sure the first example should be that of an unsafe api

File: README.md (hunk 3 above)
Comment (AdrianRaFo) [reply README.md]: How can I add Ank to be able to use it in the Readme?

Comment (raulraja) [reply README.md]: https://github.com/arrow-kt/arrow/tree/master/modules/ank

File: README.md [hunk 5]
```diff
@@ -1 +1,25 @@
-EXPERIMENTAL WIP Json library based on a port of the [Jawn Parser](https://github.com/non/jawn) built on Arrow.
+# Helios
+
+Json library based on a port of the [Jawn Parser](https://github.com/non/jawn) built on Arrow.
```
Comment (raulraja) [inline README.md]: We need a better introduction here that appeals to users better than this. //cc @maureenelsberry

File: helios-docs/build.gradle [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion"
+
+    kapt "io.arrow-kt:arrow-annotations-processor:$arrowVersion"
+    kapt project(':helios-meta')
+    kapt project(':helios-meta-compiler')
+    kapt project(':helios-dsl-syntax-compiler')
+    implementation "io.arrow-kt:arrow-ank:$arrowVersion"
+}
+
+ank {
```
Comment (raulraja) [inline helios-docs/build.gradle]: People don't need to use ank to use Helios. Why is this here?

File: helios-docs/docs/QuickStart.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+We can also encode from a data class instance to a `Json`
+
+```kotlin:ank
+val person = Person("Raul", 34)
+
+val jsonFromPerson = with(Person.encoder()) {
+  person.encode()
+}
+
+println(jsonFromPerson.toJsonString())
```
Comment (raulraja) [inline helios-docs/docs/QuickStart.md]: If you use this style of ank snippets println returns unit and the result of every expression is going to be unit

File: helios-docs/build.gradle (hunk 6 above)
Comment (AdrianRaFo) [reply helios-docs/build.gradle]: Is the build to the docs module, I created to use Ank in the docs

File: README.md [hunk 8]
```diff
@@ -1 +1,25 @@
...
+dependencies {
+    compile "com.47deg:helios-core:0.0.1-SNAPSHOT"
+    compile "com.47deg:helios-meta:0.0.1-SNAPSHOT"
+    compile "com.47deg:helios-parser:0.0.1-SNAPSHOT"
+}
+```
+
+## QuickStart
+
+You can find a quickstart [here]().
```
Comment (nomisRev) [inline README.md]:
```suggestion
You can find a quickstart [here](https://github.com/47deg/helios/tree/master/helios-docs/docs/QuickStart.md).
```

You can have it point to the file after merge until we have a website.

File: helios-docs/docs/QuickStart.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+## DSL
+
+```kotlin
+@json
+data class Person(val name: String, val age: Int) {
+  companion object
+}
+```
+
+The `@json` annotation will provide the decoder and encoder for that data class, so we are able to
```
Comment (nomisRev) [inline helios-docs/docs/QuickStart.md]: This sentence seems to be missing the end. Perhaps ", so we are able to read from and write to Json." ?

File: helios-docs/docs/QuickStart.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+}, {
+  "Successfully decode the json: $it"
+})
+```
+
+## Encode
+
+We can also encode from a data class instance to a `Json`
+
+```kotlin:ank:silent
```
Comment (nomisRev) [inline helios-docs/docs/QuickStart.md]: Why silent?

File: helios-docs/build.gradle (hunk 6 above)
Comment (raulraja) [reply helios-docs/build.gradle]: my bad, I thought this change was in the README file.

File: helios-docs/docs/QuickStart.md (hunk 10 above)
Comment (AdrianRaFo) [reply helios-docs/docs/QuickStart.md]: `silent` makes that Ank adds as a comment only the last line on the kotlin block like a result

File: README.md (hunk 5 above)
Comment (MaureenElsberry) [reply README.md]: I suggest writing a brief bit, starting with:

Helios is a Json library used to transform Json text into a model and vice versa. It's based on a part of the [Jawn Parser](https://github.com/non/jawn) built on Arrow, a Functional companion to Kotlin's Standard Library. 

Then talk a little bit about *why* people would use this? Use a sentence or two to solve the issue that the library is created for - ease of use, time, whatever it may be. 

I can edit the content when it's created - but can't really create much more than that as I'm not really deeply familiar with Helios at this point @AdrianRaFo @raulraja

**Comment by fedefernandez** [comment]
@raulraja is this looking good now? Thanks
